src/view/RepeatedFragment.js

```diff
@@ -210,6 +210,10 @@ export default class RepeatedFragment {
 			if ( newIndex === -1 ) return;
 
 			const fragment = this.iterations[ oldIndex ];
+
+			// guard against null fragment - #2005 (feels like treating the symptom, not the cause?)
+			if ( fragment == null ) return;
```

Comment:
Nice sleuthing! Definite yes to the hacky part. There's a sort of long standing TODO on tearing ArrayWrapper and MagicWrapper out into their own plugins. Magic is a different can of worms, but I think the first step to moving ArrayWrapper would be to have it delegate to the `Ractive` array methods instead of managing its own runloop turn and model. That should eliminate the `__model` hack and clear up this issue as well.

I think I have a stash somewhere that does the delegation. I can open a PR with it if you'd like, or if you'd rather dig in and handle it yourself, I'd be happy to help in whatever ways I can.